packages/bot/src/stores/toolbar-store.js

```diff
@@ -94,33 +98,38 @@ export default class ToolbarStore {
     @action.bound
     onResetOkButtonClick() {
         const workspace = Blockly.derivWorkspace;
-        Blockly.Events.setGroup('reset');
-        workspace.clear();
-        Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom(workspace.blocksXmlStr), workspace);
-        Blockly.Events.setGroup(false);
-        this.file_name = localize('Untitled Bot');
+
+        runGroupedEvents(false, () => {
+            workspace.clear();
+            Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom(workspace.blocksXmlStr), workspace);
```

Comment:
@aaron-binary you have access to all this `Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom(workspace.blocksXmlStr), workspace` why you need to pass it as an argument again? if the workspace it for some reason different, you just need to pass` Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom(workspace.blocksXmlStr), workspace` through right?

Reply:
I'm not sure what you're saying here, but I'll explain both functions. `Blockly.Xml.domToWorkspace` and `Blockly.Xml.textToDom` are static functions that require these arguments. 

- `domToWorkspace` expects an XML DOM that will be converted to a workspace and a `Blockly.Workspace` to dump that workspace to. https://developers.google.com/blockly/reference/js/Blockly.Xml#.domToWorkspace
- `textToDom` takes an XML string and converts it into an XML DOM. https://developers.google.com/blockly/reference/js/Blockly.Xml#.textToDom

These functions aren't supplied on a `Blockly.Workspace` prototype, so there's no other way (other than extending the workspace class with these functions) to achieve the same result.